### PR TITLE
feat: add version to global manifest

### PR DESCRIPTION
**Pull request by Hofer-Julian**
That way we can break while still supporting old versions

### CODE REVIEW

**Comment by ruben-arts** [comment]
image

I added that it will include the version mismatch in the error if any.

**Review by ruben-arts — APPROVED** [review]
I now approve this but it is also refactored by me, @Hofer-Julian could you give my refactor a look?

File: src/global/project/parsed_manifest.rs [hunk 1]
```diff
@@ -1,48 +1,141 @@
...
     }
 }
 
+#[derive(Error, Debug, Clone, Diagnostic)]
+pub enum ManifestParsingError {
+    #[error(transparent)]
+    Error(#[from] toml_edit::TomlError),
+    #[error("The 'version' of the manifest is too low: '{0}', the supported version is '{GLOBAL_MANIFEST_VERSION}', please update the manifest.")]
+    VersionTooLow(u32, #[source] toml_edit::TomlError),
+    #[error("The 'version' of the manifest is too high: '{0}', the supported version is '{GLOBAL_MANIFEST_VERSION}', please update `pixi` to support the new manifest version.")]
```
Comment (Hofer-Julian) [inline src/global/project/parsed_manifest.rs:47]: IIRC there shouldn't be dots at the end of error messages

File: src/global/project/parsed_manifest.rs [hunk 2]
```diff
@@ -1,31 +1,141 @@
...
+    }
+}
+
+#[derive(Error, Debug, Clone, Diagnostic)]
+pub enum ManifestParsingError {
+    #[error(transparent)]
+    Error(#[from] toml_edit::TomlError),
+    #[error("The 'version' of the manifest is too low: '{0}', the supported version is '{GLOBAL_MANIFEST_VERSION}', please update the manifest.")]
+    VersionTooLow(u32, #[source] toml_edit::TomlError),
+    #[error("The 'version' of the manifest is too high: '{0}', the supported version is '{GLOBAL_MANIFEST_VERSION}', please update `pixi` to support the new manifest version.")]
```
Comment (ruben-arts) [inline src/global/project/parsed_manifest.rs:47]:
```suggestion
    #[error("The 'version' of the manifest is too low: '{0}', the supported version is '{GLOBAL_MANIFEST_VERSION}', please update the manifest")]
    VersionTooLow(u32, #[source] toml_edit::TomlError),
    #[error("The 'version' of the manifest is too high: '{0}', the supported version is '{GLOBAL_MANIFEST_VERSION}', please update `pixi` to support the new manifest version")]
```

**Comment by Hofer-Julian** [comment]
This is ready to be merged from my side

**Pull request merged**